public/app/plugins/datasource/influxdb/datasource.ts

```diff
@@ -345,26 +347,39 @@ export default class InfluxDatasource extends DataSourceWithBackend<InfluxQuery,
 
   // By implementing getTagKeys and getTagValues we add ad-hoc filters functionality
   // Used in public/app/features/variables/adhoc/picker/AdHocFilterKey.tsx::fetchFilterKeys
-  getTagKeys(options: InfluxQuery) {
+  getTagKeys(options: DataSourceGetTagKeysOptions) {
```

Comment:
In the `@grafana-data/src/types/datasource.ts` (parent class) this method is defined with an `optional` options parameter. We should have the same here too.

Reply:
@itsmylife good point, I missed that. I would love to require the argument but since this can technically be called from plugins I am torn on what to do. We can make sure Grafana always calls it with an argument but not if other plugin code do